internal/pkg/action/dispatcher.go

```diff
@@ -96,31 +97,63 @@ func (d *Dispatcher) process(ctx context.Context, hits []es.HitT) {
 			log.Error().Err(err).Msg("Failed to unmarshal action document")
 			break
 		}
-		for _, agentId := range action.Agents {
-			arr := agentActions[agentId]
+		numAgents := len(action.Agents)
+		for i, agentID := range action.Agents {
+			arr := agentActions[agentID]
 			actionNoAgents := action
+			actionNoAgents.StartTime = offsetStartTime(action.StartTime, action.Expiration, action.MinimumExecutionDuration, i, numAgents)
 			actionNoAgents.Agents = nil
 			arr = append(arr, actionNoAgents)
-			agentActions[agentId] = arr
+			agentActions[agentID] = arr
 		}
 	}
 
-	for agentId, actions := range agentActions {
-		d.dispatch(ctx, agentId, actions)
+	for agentID, actions := range agentActions {
+		d.dispatch(ctx, agentID, actions)
 	}
 }
 
-func (d *Dispatcher) getSub(agentId string) (Sub, bool) {
+// offsetStartTime will return a new start time between start:end-dur based on index i and the total number of agents
+// An empty string will be returned if start or exp are empty or if there is an error parsing inputs
+// As we expect i < total  the latest return time will always be < exp-dur
+func offsetStartTime(start, exp, dur string, i, total int) string {
+	if start == "" || exp == "" {
+		return ""
+	}
+	startTS, err := time.Parse(time.RFC3339, start) // TODO what format does a date-time string use?
```

Comment:
Are you asking which is the format used when converting a time.Time to JSON? Or that it needed to double check with the agent to ensure the're using the same format?

Reply:
I'm not sure what format Elasticsearch serializes date attributes as, that's what the TODO was for.

Reply:
IIRC it's RFC3339